### PR TITLE
feat(serialize_meta_details): return stream and video progress

### PR DESCRIPTION
**Pull request by tymmesyde**
Related: https://github.com/Stremio/stremio-core/pull/461

### CODE REVIEW

File: src/model/serialize_meta_details.rs [hunk 1]
```diff
@@ -39,6 +39,7 @@ mod model {
     pub struct Stream<'a> {
         #[serde(flatten)]
         pub stream: &'a stremio_core::types::resource::Stream,
+        pub progress: Option<f64>,
```
Comment (elpiel) [inline src/model/serialize_meta_details.rs:44]: - [x] Please document the field - What is this progress? Percentage, time, ms, apples?!

File: src/model/serialize_meta_details.rs [hunk 2]
```diff
@@ -177,6 +185,7 @@ pub fn serialize_meta_details(
                             .iter()
                             .map(|stream| model::Stream {
                                 stream,
+                                progress: None,
```
Comment (elpiel) [inline src/model/serialize_meta_details.rs:190]: Can we track progress on trailers?
If they use the same streaming bucket I suppose we can?!

Comment (tymmesyde) [reply src/model/serialize_meta_details.rs:190]: We don't show a list of trailers in our apps we only play the first available one

**Comment by elpiel** [comment]
stremio-web works as expected and shows the progress of a Libraryitem:
Screenshot from 2023-09-13 14-00-30

Remaining todos:
- [ ] Update core after merge!

**Pull request merged**